mllib-dal/src/main/scala/com/intel/oap/mllib/OneDAL.scala

```diff
@@ -651,15 +658,56 @@ object OneDAL {
 
   def coalesceToHomogenTables(data: RDD[Vector], executorNum: Int,
                                 device: Common.ComputeDevice): RDD[Long] = {
+    logger.info(s"Processing partitions with $executorNum executors")
+    val numberCores: Int  = data.sparkContext.getConf.getInt("spark.executor.cores", 0)
```

Comment:
Use 1 instead, using 0 will cause potential divided by 0 error below. and 0 makes no sense to numberCores.